# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/fping/hotfix-proto.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- fping-5.3/src/socket6.c.vanilla	2025-06-29 01:28:17.949312355 +0200
+++ fping-5.3/src/socket6.c	2025-06-29 01:28:29.812928163 +0200
@@ -52,8 +52,8 @@
     int s;
 
     /* confirm that ICMP6 is available on this machine */
-    if ((proto = getprotobyname("ipv6-icmp")) == NULL)
-        crash_and_burn("ipv6-icmp: unknown protocol");
+    if ((proto = getprotobyname("icmp6")) == NULL && (proto = getprotobyname("ipv6-icmp")) == NULL)
+        crash_and_burn("icmp6: unknown protocol");
 
     /* create raw socket for ICMP6 calls (ping) */
     *socktype = SOCK_RAW;
